### DISCUSSION TITLE
Create API to allow other services and UIs to get list of models, etc.

### DISCUSSION THREAD

**Discussion by JustMaier** [civitai/civitai#13] [Ideas]
The concept here is pretty straightforward. Let's give each user an API token and create an API that allows them to do the following:
- [x] Get a list of models with the following filters (tags, user, keyword, rating, verified, model hash)

The hope is that we could then allow them to use this API to pull a list of models into their UI and download them on demand.

**Comment by GabrielVidal1** [comment]
Any news on this ? It's a real hassle to download models programmaticlly on a distant server, a read API token like huggingface would be great!
A cli to do all this would be soooo useful

**Comment by JustMaier** [comment]
We're working on this now. @manuelurenah is trying to determine the best way to do API tokens on top of our existing auth system, NextAuth.

If it looks like that's going to take too long, it might make sense to at least create the read endpoints.

**Comment by manuelurenah** [comment]
@JustMaier @GabrielVidal1 To update this discussion, we have released a public REST API endpoint to get the models we have stored in our database along with their versions. Each version includes a download url so  you can get the model files more easily. This is obviously in active development and we will continue to make improvements along the way. In the meantime, you can checkout the wiki page for an in-depth overview of the API so far https://github.com/civitai/civitai/wiki/REST-API-Reference
